source/megengine.js

```diff
@@ -3,30 +3,96 @@
 
 var megengine = megengine || {};
 var base = base || require('./base');
+var zip = zip || require('./zip');
 
 megengine.ModelFactory = class {
 
     match(context) {
+        megengine.modelType = '';
+        const identifier = context.identifier;
+        const extension = identifier.split('.').pop().toLowerCase();
+        const stream = context.stream;
+        if (extension === 'mge' && stream.length >= 8) {
+            const buffer = stream.peek(Math.min(32, stream.length));
+            if (buffer.length >= 12) {
+                const magic = String.fromCharCode(...buffer.slice(8, 12));
+                if (magic === 'mgv2') {
+                    megengine.modelType = 'megengine.flatbuffers';
+                    return megengine.modelType;
+                }
+                throw new megengine.Error('Only support megengine c++ model v2');
+            }
+        }
         const obj = context.open('pkl');
         if (obj.__class__ && obj.__class__.__module__ === 'megengine.traced_module.traced_module' && obj.__class__.__name__ === 'TracedModule') {
-            return 'megengine.pickle';
+            megengine.modelType = 'megengine.pickle';
         }
-        return '';
+        return megengine.modelType;
     }
 
     open(context) {
         return context.metadata('megengine-metadata.json').then((metadata) => {
-            const obj = context.open('pkl');
-            return new megengine.Model(metadata, obj);
+            switch (megengine.modelType) {
+                case 'megengine.pickle': {
+                    const obj = context.open('pkl');
+                    return new megengine.Model(metadata, obj);
+                }
+                case 'megengine.flatbuffers': {
+                    return context.require('./megengine-schema').then(() => {
+                        megengine.schema_main = flatbuffers.get('megengine').mgb.serialization.fbs;
```

Comment:
Use `megengine.schema` and full namespaces to make it easier to search for all Flatbuffers namespaces.

Reply:
done